pkgs/by-name/se/semaphoreui/package.nix

```
@@ -0,0 +1,69 @@
+{
+  lib,
+  stdenv,
+  fetchurl,
+  gnutar,
+}:
+
+let
+  platformInfo = {
+    "x86_64-linux" = {
+      arch = "linux_amd64";
+      sha256 = "sha256-gBdeN06fDcFP37+Igw7YGMR9Qy9kP8wMd/lqgrEEHEo=";
+    };
+    "aarch64-linux" = {
+      arch = "linux_arm64";
+      sha256 = "sha256-Byizs9nIX2eyUGq+1bJ3OJnyk5B3EHh2/n5DYgNIiPY=";
+    };
+    "x86_64-darwin" = {
+      arch = "darwin_amd64";
+      sha256 = "sha256-RYbygr3vrFIbvc1edm4FcEG4fHCMP5/zWC+E99zyfQY=";
+    };
+    "aarch64-darwin" = {
+      arch = "darwin_arm64";
+      sha256 = "sha256-C0TBr98pzIClC54P8itAR7oEK0msi3JwW9YKD2EHhRI=";
+    };
+  };
+
+  platform =
+    platformInfo.${stdenv.hostPlatform.system}
+      or (throw "Unsupported system: ${stdenv.hostPlatform.system}");
+
+in
+stdenv.mkDerivation rec {
+  pname = "semaphoreui";
+  version = "2.15.0";
+
+  src = fetchurl {
+    url = "https://github.com/semaphoreui/semaphore/releases/download/v${version}/semaphore_${version}_${platform.arch}.tar.gz";
+    sha256 = platform.sha256;
+  };
+
+  nativeBuildInputs = [ gnutar ];
+
+  unpackPhase = ''
+    tar xf $src
+  '';
+
+  installPhase = ''
+    mkdir -p $out/bin
+    cp semaphore $out/bin/
+    chmod +x $out/bin/semaphore
+  '';
```

```suggestion
  installPhase = ''
    runHook preInstall
    install -Dm555 semaphore -t $out/bin/
    runHook postInstall
  '';
```
1. The phases must remain adjustable, hence `runHook`.
2. `install -DmNNN SOURCE... -t PATH` is shorter and is just more common in Nixpkgs.